akka-docs-new/src/test/scala/tutorial_3/DeviceGroup.scala

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package tutorial_3
+
+import akka.actor.{ Actor, ActorLogging, ActorRef, Cancellable, Props, Stash, Terminated }
+import tutorial_3.DeviceGroup._
+import tutorial_3.DeviceManager.RequestTrackDevice
+import scala.concurrent.duration._
+
+object DeviceGroup {
+
+  def props(groupId: String): Props = Props(new DeviceGroup(groupId))
+
+  case class RequestDeviceList(requestId: Long)
+  case class ReplyDeviceList(requestId: Long, ids: Set[String])
+
+  case class RequestAllTemperatures(requestId: Long)
+  case class RespondAllTemperatures(requestId: Long, temperatures: Map[String, Option[Double]])
```

Comment:
I have a bit richer model now, not using option but various values representing if there is a reading, or the device is there but no reading is available, the device is no longer there, or if the device did not respond in time.